src/layers/yarn_scaled_rotary_embedding.cpp

```diff
@@ -160,22 +113,22 @@ void LlamaYaRNScaledRotaryEmbedding::forward(
         for (int bs = 0; bs < batchSize; ++bs) {
             for (int seq = 0; seq < seqLen; ++seq) {
                 int pos = positionIds[seq];
-                float *pcos = embCos + pos * dim;
-                float *psin = embSin + pos * dim;
+                float *pcos = embCos + pos * half;
```

Comment:
Original code is:
float *pcos = embCos + pos * dim;
are we sure need to change to current code?
float *pcos = embCos + pos * half;

Reply:
Previously, the lens of pcos and psin were “dim”, with the first half and the second half being same. Now, pcos and psin have been unified to inv_freq_size (dim / 2), so we made the change here.